### PR TITLE
Fix #713 : Lowfi Landscape Topic Practice

### LINKED ISSUE #713: Lowfi Landscape Topic Practice
Mock Link: https://xd.adobe.com/view/ee9e607b-dbd6-4372-48dc-b687d32af3da-98af/screen/0fa47d7c-e10d-4097-8773-0efa665a323f/TP-Practice-

## Checklist
- [ ] Landscape lowfi mock
- [ ] Test cases

### PR DESCRIPTION
**Pull request by Sarthak2601**
## Explanation

- Fix #713 
- Landscape mode of Topic practice according to the mock. 
- Implemented tests. 

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: app/src/main/res/layout-land/topic_practice_footer_view.xml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      android:clickable="@{isSubmitButtonActive}"
+      android:minHeight="36dp"
+      android:text="@string/topic_practice_start"
+      android:textColor="@{isSubmitButtonActive ? @color/white : @color/submitButtonInactive}"
+      app:layout_constraintEnd_toEndOf="parent"
+      app:layout_constraintStart_toStartOf="parent"
+      app:layout_constraintTop_toTopOf="parent"
+      app:layout_constraintBottom_toBottomOf="parent"/>
+  </androidx.constraintlayout.widget.ConstraintLayout>
+</layout>
```
Comment (rt4914) [inline app/src/main/res/layout-land/topic_practice_footer_view.xml]: This file is missing EOF.

File: app/src/main/res/layout-land/topic_practice_footer_view.xml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    <variable
+      name="isSubmitButtonActive"
+      type="Boolean" />
+
+    <variable
+      name="viewModel"
+      type="org.oppia.app.topic.practice.practiceitemviewmodel.TopicPracticeFooterViewModel"/>
+  </data>
+  <androidx.constraintlayout.widget.ConstraintLayout
```
Comment (rt4914) [inline app/src/main/res/layout-land/topic_practice_footer_view.xml:16]: The indentation in this file is incorrect.
Press `Control + Alt + L` to fix indentation or check other files to fix that.

File: app/src/main/res/layout-land/topic_practice_fragment.xml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    android:paddingBottom="172dp"
+    android:scrollbars="none"
+    android:paddingStart="48dp"
+    android:paddingEnd="48dp"
+    app:data="@{viewModel.topicPracticeSkillLiveData}"
+    app:layout_constraintBottom_toBottomOf="parent"
+    app:layout_constraintEnd_toEndOf="parent"
+    app:layout_constraintStart_toStartOf="parent"
+    app:layout_constraintTop_toTopOf="parent"/>
+</layout>
```
Comment (rt4914) [inline app/src/main/res/layout-land/topic_practice_fragment.xml]: Ditto as above file, EOF is missing and indentation is incorrect.

File: app/src/main/res/layout-land/topic_practice_header_view.xml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      android:layout_height="wrap_content"
+      android:layout_marginTop="8dp"
+      android:fontFamily="sans-serif"
+      android:text="@string/topic_practice_skills_description"
+      android:textColor="@color/oppiaSecondaryText"
+      android:textSize="16sp"
+      app:layout_constraintStart_toStartOf="@+id/master_skills_text_view"
+      app:layout_constraintTop_toBottomOf="@+id/master_skills_text_view"/>
+  </androidx.constraintlayout.widget.ConstraintLayout>
+</layout>
```
Comment (rt4914) [inline app/src/main/res/layout-land/topic_practice_header_view.xml]: Ditto as above file.

File: app/src/main/res/layout-land/topic_practice_subtopic.xml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+      android:paddingEnd="20dp"
+      android:text="@{viewModel.subtopic.title}"
+      android:textColor="@color/oppiaPrimaryText"
+      android:textSize="16sp"
+      app:layout_constraintBottom_toBottomOf="parent"
+      app:layout_constraintEnd_toEndOf="parent"
+      app:layout_constraintStart_toStartOf="parent"
+      app:layout_constraintTop_toTopOf="parent" />
+  </androidx.constraintlayout.widget.ConstraintLayout>
+</layout>
```
Comment (rt4914) [inline app/src/main/res/layout-land/topic_practice_subtopic.xml]: Ditto as above file.

File: app/src/sharedTest/java/org/oppia/app/topic/practice/TopicPracticeFragmentTest.kt [hunk 6]
```diff
@@ -160,38 +163,64 @@ class TopicPracticeFragmentTest {
...
+    launchTopicActivityIntent(FRACTIONS_TOPIC_ID).use {
+      onView(
+        allOf(
+          withText(TopicTab.getTabForPosition(2).name),
+          isDescendantOfA(withId(R.id.topic_tabs_container))
+        )
+      ).perform(click())
+      onView(withId(R.id.master_skills_text_view)).check(matches(withText(R.string.topic_practice_master_these_skills)))
+      onView(isRoot()).perform(orientationLandscape())
+      onView(withId(R.id.master_skills_text_view)).check(matches(withText(R.string.topic_practice_master_these_skills)))
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/topic/practice/TopicPracticeFragmentTest.kt:271]: Press `Control + Options + L` to fix indentation. Also, this file contains some unused imports.

File: app/src/sharedTest/java/org/oppia/app/topic/practice/TopicPracticeFragmentTest.kt [hunk 7]
```diff
@@ -160,38 +163,64 @@ class TopicPracticeFragmentTest {
...
+        )
+      ).perform(click())
+      onView(atPositionOnView(R.id.topic_practice_skill_list, 1, R.id.subtopic_check_box)).perform(click())
+      onView(isRoot()).perform(orientationLandscape())
+      onView(withId(R.id.topic_practice_skill_list)).perform(scrollToPosition<RecyclerView.ViewHolder>(10))
+      onView(withId(R.id.topic_practice_start_button)).check(matches(isClickable()))    }
+  }
+
+  @Test
+  fun testTopicPracticeFragment_loadFragment_changeorientation_titleiscorrect() {
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/topic/practice/TopicPracticeFragmentTest.kt]: Name is incorrect. We are following camelCase convention separated by `_`. That means that `changeorientation` should be `changeOrientation` and `titleiscorrect` should be `titleIsCorrect`.
`testTopicPracticeFragment_loadFragment_changeOrientation_titleIsCorrect`

**Review by rt4914 — CHANGES_REQUESTED** [review]
Looks good but there are some basic changes. Also, please make sure that once you create a PR, you always self review it by checking the code in `Files Changed` tab. This helps make your PR approval faster.

File: app/src/main/res/layout-land/topic_practice_footer_view.xml (hunk 1 above)
Comment (Sarthak2601) [reply app/src/main/res/layout-land/topic_practice_footer_view.xml]: Done:)

File: app/src/main/res/layout-land/topic_practice_fragment.xml (hunk 3 above)
Comment (Sarthak2601) [reply app/src/main/res/layout-land/topic_practice_fragment.xml]: Done:)

File: app/src/main/res/layout-land/topic_practice_subtopic.xml (hunk 5 above)
Comment (Sarthak2601) [reply app/src/main/res/layout-land/topic_practice_subtopic.xml]: Done:)

File: app/src/sharedTest/java/org/oppia/app/topic/practice/TopicPracticeFragmentTest.kt (hunk 7 above)
Comment (Sarthak2601) [reply app/src/sharedTest/java/org/oppia/app/topic/practice/TopicPracticeFragmentTest.kt]: Done:)

File: app/src/main/res/layout-land/topic_practice_header_view.xml (hunk 4 above)
Comment (Sarthak2601) [reply app/src/main/res/layout-land/topic_practice_header_view.xml]: Done:)

File: app/src/sharedTest/java/org/oppia/app/topic/practice/TopicPracticeFragmentTest.kt (hunk 6 above)
Comment (Sarthak2601) [reply app/src/sharedTest/java/org/oppia/app/topic/practice/TopicPracticeFragmentTest.kt:271]: When I do that, there are many functions that become multi-line but were single-line previously. Should I proceed with `Control+Options+L` ?

File: app/src/main/res/layout-land/topic_practice_footer_view.xml (hunk 2 above)
Comment (Sarthak2601) [reply app/src/main/res/layout-land/topic_practice_footer_view.xml:16]: This file exactly matches its portrait mode copy in terms of indentation. Should I proceed with  `Control+Alt+L` ?

File: app/src/sharedTest/java/org/oppia/app/topic/practice/TopicPracticeFragmentTest.kt (hunk 6 above)
Comment (rt4914) [reply app/src/sharedTest/java/org/oppia/app/topic/practice/TopicPracticeFragmentTest.kt:271]: Yes do proceed with it, if there are some unnecessary changes introduced I will check that in another review.

File: app/src/main/res/layout-land/topic_practice_footer_view.xml (hunk 2 above)
Comment (rt4914) [reply app/src/main/res/layout-land/topic_practice_footer_view.xml:16]: Yes, and fix that in portrait mode also.

Comment (Sarthak2601) [reply app/src/main/res/layout-land/topic_practice_footer_view.xml:16]: Done PTAL.

File: app/src/sharedTest/java/org/oppia/app/topic/practice/TopicPracticeFragmentTest.kt (hunk 6 above)
Comment (Sarthak2601) [reply app/src/sharedTest/java/org/oppia/app/topic/practice/TopicPracticeFragmentTest.kt:271]: Done PTAL.

File: app/src/main/res/layout-land/topic_practice_fragment.xml [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    android:paddingStart="48dp"
+    android:paddingEnd="48dp"
+    android:paddingBottom="172dp"
+    android:scrollbars="none"
+    app:data="@{viewModel.topicPracticeSkillLiveData}"
+    app:layout_constraintBottom_toBottomOf="parent"
+    app:layout_constraintEnd_toEndOf="parent"
+    app:layout_constraintStart_toStartOf="parent"
+    app:layout_constraintTop_toTopOf="parent" />
+</layout>
```
Comment (rt4914) [inline app/src/main/res/layout-land/topic_practice_fragment.xml:27]: `Ctrl + Alt + L` missing in this file too.

File: app/src/main/res/layout-land/topic_practice_header_view.xml [hunk 9]
```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="utf-8"?>
+<layout xmlns:android="http://schemas.android.com/apk/res/android"
+        xmlns:app="http://schemas.android.com/apk/res-auto">
+  <data>
+    <variable
+      name="viewModel"
+      type="org.oppia.app.topic.practice.practiceitemviewmodel.TopicPracticeHeaderViewModel" />
+  </data>
```
Comment (rt4914) [inline app/src/main/res/layout-land/topic_practice_header_view.xml:10]: `Ctrl + Alt + L` missing in this file too.

File: app/src/sharedTest/java/org/oppia/app/topic/practice/TopicPracticeFragmentTest.kt [hunk 10]
```diff
@@ -152,50 +178,103 @@ class TopicPracticeFragmentTest {
...
     intended(hasExtra(QuestionPlayerActivity.getIntentKey(), skillIdList))
   }
 
   @Test
-  @Ignore("Landscape not properly supported") // TODO(#56): Reenable once landscape is supported.
-  fun testTopicPracticeFragment_loadFragment_selectSkills_configurationChange_skillsAreSelected() {
-    onView(atPositionOnView(R.id.topic_practice_skill_list, 1, R.id.subtopic_check_box)).perform(click())
-    activityScenario.onActivity { activity ->
-      activity.requestedOrientation = Configuration.ORIENTATION_LANDSCAPE
+  fun testTopicPracticeFragment_loadFragment_selectSkills_configurationChange_skillsAreSelect() {
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/topic/practice/TopicPracticeFragmentTest.kt]: Change the last word
`testTopicPracticeFragment_loadFragment_selectSkills_configurationChange_skillsAreSelected`

**Review by rt4914 — CHANGES_REQUESTED** [review]
Nit changes suggested.

File: app/src/sharedTest/java/org/oppia/app/topic/practice/TopicPracticeFragmentTest.kt (hunk 10 above)
Comment (Sarthak2601) [reply app/src/sharedTest/java/org/oppia/app/topic/practice/TopicPracticeFragmentTest.kt]: Done

File: app/src/main/res/layout-land/topic_practice_header_view.xml (hunk 9 above)
Comment (Sarthak2601) [reply app/src/main/res/layout-land/topic_practice_header_view.xml:10]: Done

File: app/src/main/res/layout-land/topic_practice_fragment.xml (hunk 8 above)
Comment (Sarthak2601) [reply app/src/main/res/layout-land/topic_practice_fragment.xml:27]: Done

**Review by rt4914 — COMMENTED** [review]
LGTM, thanks.

**Comment by Sarthak2601** [comment]
> LGTM, thanks.

Oh great ! Thanks a lot for the help :)

**Review by rt4914 — APPROVED** [review]
LGTM

**Pull request merged**